test/test_cache.py

```diff
@@ -170,6 +170,13 @@ def test_update_check_consumer_uuid_none(self):
         res = self.profile_mgr.update_check(uep, uuid)
         self.assertEqual(0, res)
 
+    def test_package_json_handles_non_unicode(self):
+        package = Package(name="package1", version="1.0.0", release=1, arch="x86_64", vendor=b'\xf6')
+        data = package.to_dict()
+        json_str = json.dumps(data)  # to json
+        data = json.loads(json_str)  # and back to an object
+        self.assertEqual(u'\ufffd', data['vendor'])
```

Comment:
I would add more tests here (e.g. with `vendor='Borg'`, `vendor='Björk'`) due to my previous comment.